django_date_extensions/fields.py

```diff
@@ -114,12 +139,28 @@ def to_python(self, value):
         if value == 'past':
             return ApproximateDate(past=True)
 
-        if not ansi_date_re.search(value):
-            raise ValidationError('Enter a valid date in YYYY-MM-DD format.')
+        prefix = None
+
+        if isinstance(value, (datetime.date, datetime.datetime)):
+            year, month, day = value.year, value.month, value.day
+        else:
+            prefix_date = prefix_date_re.search(value)
+            prefix_date_reverse = prefix_date_reverse_re.search(value)
```

Comment:
In database the prefix dates are stored in reverse e.g. if user will input `about 2016` the database will hold value `2016 about` this was done to support correct ordering see [ordering test](https://github.com/intellisense/django-date-extensions/blob/412039601600dacda2f78c6e227a43b5dd7c5914/django_date_extensions/tests.py#L225)